spring-boot-starter-camunda-sdk will run into a NPE if legacy 8.5 properties are used

**Describe the bug**

When not using the new 8.6 [`camunda.client.*` properties](https://docs.camunda.io/docs/apis-tools/spring-zeebe-sdk/getting-started/#self-managed) but relying on legacy 8.5 [`zeebe.client.*` ](https://docs.camunda.io/docs/8.5/apis-tools/spring-zeebe-sdk/getting-started/#configuring-the-zeebe-cluster-connection) properties the zeebe client will fail invoking commands due to a NPE when checking whether to prefer REST over GRPC.

This was introduced by this change https://github.com/camunda/camunda/commit/0779e115a6e79aa273c3215bb202bf98bb504c26 that made command execution dependent on the `preferRestOverGrpc` field.

**To Reproduce**

1. Use legacy 8.5 [`zeebe.client.*` ](https://docs.camunda.io/docs/8.5/apis-tools/spring-zeebe-sdk/getting-started/#configuring-the-zeebe-cluster-connection) to configure the spring sdk
2. Update spring-boot-starter-camunda-sdk to 8.6

See this fix from client side perspective https://github.com/camunda/camunda-8-get-started-spring/pull/9/files#diff-54eeffbae371fcd1398d4ca5e89a1b8118208b7bb2f8ddf55c1aa2f7d98ab136

**Expected behavior**

Old properties setups should still work.

**Workaround**

Migrate to new 8.6 [`camunda.client.*` properties](https://docs.camunda.io/docs/apis-tools/spring-zeebe-sdk/getting-started/#self-managed)

**Log/Stacktrace**

<!-- If possible add the full stacktrace or Zeebe log which contains the issue. -->

<details><summary>Full Stacktrace</summary>
 <p>

```
org.springframework.context.ApplicationContextException: Failed to start bean 'zeebeLifecycleEventProducer'
	at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:291) ~[spring-context-6.1.8.jar:6.1.8]
	at org.springframework.context.support.DefaultLifecycleProcessor$LifecycleGroup.start(DefaultLifecycleProcessor.java:471) ~[spring-context-6.1.8.jar:6.1.8]
	at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[na:na]
	at org.springframework.context.support.DefaultLifecycleProcessor.startBeans(DefaultLifecycleProcessor.java:260) ~[spring-context-6.1.8.jar:6.1.8]
	at org.springframework.context.support.DefaultLifecycleProcessor.onRefresh(DefaultLifecycleProcessor.java:205) ~[spring-context-6.1.8.jar:6.1.8]
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:981) ~[spring-context-6.1.8.jar:6.1.8]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:627) ~[spring-context-6.1.8.jar:6.1.8]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754) ~[spring-boot-3.3.0.jar:3.3.0]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:456) ~[spring-boot-3.3.0.jar:3.3.0]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:335) ~[spring-boot-3.3.0.jar:3.3.0]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1363) ~[spring-boot-3.3.0.jar:3.3.0]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1352) ~[spring-boot-3.3.0.jar:3.3.0]
	at io.camunda.demo.process_payments.ProcessPaymentsApplication.main(ProcessPaymentsApplication.java:25) ~[classes/:na]
Caused by: java.lang.NullPointerException: Cannot invoke "io.camunda.zeebe.spring.client.properties.common.ZeebeClientProperties.isPreferRestOverGrpc()" because the return value of "io.camunda.zeebe.spring.client.properties.CamundaClientProperties.getZeebe()" is null
	at io.camunda.zeebe.spring.client.configuration.ZeebeClientConfigurationImpl.preferRestOverGrpc(ZeebeClientConfigurationImpl.java:329) ~[spring-boot-starter-camunda-sdk-8.6.0.jar:8.6.0]
	at io.camunda.zeebe.client.impl.ZeebeClientImpl.newDeployResourceCommand(ZeebeClientImpl.java:364) ~[zeebe-client-java-8.6.0.jar:8.6.0]
	at io.camunda.zeebe.spring.client.annotation.processor.ZeebeDeploymentAnnotationProcessor.lambda$start$6(ZeebeDeploymentAnnotationProcessor.java:66) ~[spring-boot-starter-camunda-sdk-8.6.0.jar:8.6.0]
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596) ~[na:na]
	at io.camunda.zeebe.spring.client.annotation.processor.ZeebeDeploymentAnnotationProcessor.start(ZeebeDeploymentAnnotationProcessor.java:63) ~[spring-boot-starter-camunda-sdk-8.6.0.jar:8.6.0]
	at io.camunda.zeebe.spring.client.annotation.processor.ZeebeAnnotationProcessorRegistry.lambda$startAll$0(ZeebeAnnotationProcessorRegistry.java:54) ~[spring-boot-starter-camunda-sdk-8.6.0.jar:8.6.0]
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596) ~[na:na]
	at io.camunda.zeebe.spring.client.annotation.processor.ZeebeAnnotationProcessorRegistry.startAll(ZeebeAnnotationProcessorRegistry.java:54) ~[spring-boot-starter-camunda-sdk-8.6.0.jar:8.6.0]
	at io.camunda.zeebe.spring.client.event.ZeebeClientEventListener.handleStart(ZeebeClientEventListener.java:32) ~[spring-boot-starter-camunda-sdk-8.6.0.jar:8.6.0]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[na:na]
	at org.springframework.context.event.ApplicationListenerMethodAdapter.doInvoke(ApplicationListenerMethodAdapter.java:365) ~[spring-context-6.1.8.jar:6.1.8]
	at org.springframework.context.event.ApplicationListenerMethodAdapter.processEvent(ApplicationListenerMethodAdapter.java:237) ~[spring-context-6.1.8.jar:6.1.8]
	at org.springframework.context.event.ApplicationListenerMethodAdapter.onApplicationEvent(ApplicationListenerMethodAdapter.java:168) ~[spring-context-6.1.8.jar:6.1.8]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:185) ~[spring-context-6.1.8.jar:6.1.8]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:178) ~[spring-context-6.1.8.jar:6.1.8]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:156) ~[spring-context-6.1.8.jar:6.1.8]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:451) ~[spring-context-6.1.8.jar:6.1.8]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:384) ~[spring-context-6.1.8.jar:6.1.8]
	at io.camunda.zeebe.spring.client.event.ZeebeLifecycleEventProducer.start(ZeebeLifecycleEventProducer.java:38) ~[spring-boot-starter-camunda-sdk-8.6.0.jar:8.6.0]
	at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:288) ~[spring-context-6.1.8.jar:6.1.8]
	... 12 common frames omitted
```

</p>
</details>

**Environment:**
- OS: <!-- [e.g. Linux] -->
- Zeebe Version: <!-- [e.g. 0.20.0] --> 8.6.0
- Configuration: <!-- [e.g. exporters etc.] --> using 8.5 spring sdk config properties

--- a/clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/configuration/ZeebeClientConfigurationImpl.java
+++ b/clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/configuration/ZeebeClientConfigurationImpl.java
@@ -326,7 +326,11 @@ public class ZeebeClientConfigurationImpl implements ZeebeClientConfiguration {
 
   @Override
   public boolean preferRestOverGrpc() {
-    return camundaClientProperties.getZeebe().isPreferRestOverGrpc();
+    return getOrDefault(
+        "preferRestOverGrpc",
+        () -> camundaClientProperties.getZeebe().isPreferRestOverGrpc(),
+        DEFAULT.preferRestOverGrpc(),
+        configCache);
   }
 
   private CredentialsProvider credentialsProvider() {

--- a/clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/ZeebeClientConfigurationDefaultPropertiesTest.java
+++ b/clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/ZeebeClientConfigurationDefaultPropertiesTest.java
@@ -75,5 +75,6 @@ public class ZeebeClientConfigurationDefaultPropertiesTest {
     assertThat(client.getConfiguration().getOverrideAuthority()).isNull();
     assertThat(client.getConfiguration().getRestAddress())
         .isEqualTo(new URI("https://0.0.0.0:8080"));
+    assertThat(client.getConfiguration().preferRestOverGrpc()).isFalse();
   }
 }

--- a/clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/ZeebeClientConfigurationImplSaasTest.java
+++ b/clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/ZeebeClientConfigurationImplSaasTest.java
@@ -195,4 +195,10 @@ public class ZeebeClientConfigurationImplSaasTest {
     assertThat(output).contains("clientId='***'");
     assertThat(output).contains("clientSecret='***'");
   }
+
+  @Test
+  void shouldHaveDefaultPreferRestOverGrpc() {
+    assertThat(zeebeClientConfiguration.preferRestOverGrpc())
+        .isEqualTo(DEFAULT.preferRestOverGrpc());
+  }
 }

--- a/clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/ZeebeClientConfigurationImplSelfManagedTest.java
+++ b/clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/ZeebeClientConfigurationImplSelfManagedTest.java
@@ -183,4 +183,10 @@ public class ZeebeClientConfigurationImplSelfManagedTest {
     assertThat(zeebeClientConfiguration.useDefaultRetryPolicy())
         .isEqualTo(DEFAULT.useDefaultRetryPolicy());
   }
+
+  @Test
+  void shouldHaveDefaultPreferRestOverGrpc() {
+    assertThat(zeebeClientConfiguration.preferRestOverGrpc())
+        .isEqualTo(DEFAULT.preferRestOverGrpc());
+  }
 }

--- a/clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/legacy/ZeebeClientStarterAutoConfigurationCustomJsonMapperTest.java
+++ b/clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/legacy/ZeebeClientStarterAutoConfigurationCustomJsonMapperTest.java
@@ -117,6 +117,7 @@ public class ZeebeClientStarterAutoConfigurationCustomJsonMapperTest {
     assertThat(client.getConfiguration().getDefaultJobWorkerMaxJobsActive()).isEqualTo(99);
     assertThat(client.getConfiguration().getDefaultJobPollInterval())
         .isEqualTo(Duration.ofSeconds(99));
+    assertThat(client.getConfiguration().preferRestOverGrpc()).isFalse();
   }
 
   @EnableConfigurationProperties(CamundaClientProperties.class)

--- a/clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/legacy/ZeebeClientStarterAutoConfigurationTest.java
+++ b/clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/zeebe/spring/client/config/legacy/ZeebeClientStarterAutoConfigurationTest.java
@@ -110,6 +110,7 @@ public class ZeebeClientStarterAutoConfigurationTest {
     assertThat(client.getConfiguration().getDefaultJobWorkerMaxJobsActive()).isEqualTo(99);
     assertThat(client.getConfiguration().getDefaultJobPollInterval())
         .isEqualTo(Duration.ofSeconds(99));
+    assertThat(client.getConfiguration().preferRestOverGrpc()).isFalse();
   }
 
   @EnableConfigurationProperties(CamundaClientProperties.class)

